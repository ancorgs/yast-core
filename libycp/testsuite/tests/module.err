Parsed:
----------------------------------------------------------------------
{
  module "NameOfModule";
  global integer `ag = 1;
  integer `al = 2;
  global define NameOfModule() ``{
  y2milestone ("I am a constructor\n");
  `al = (5);
}
;
  global define af() ``{
  return (ag + al);
}
;
}
----------------------------------------------------------------------
[YCP] tests/module.ycp:8 I am a constructor
Parsed:
----------------------------------------------------------------------
{
  _dump_scope ("NameOfModule");
  return NameOfModule::af ();
}
----------------------------------------------------------------------
-- level 0 --
   NameOfModule()                 = {
  y2milestone ("I am a constructor\n");
  `al = (5);
}
   af()                     = {
  return (ag + al);
}
   integer  ag              = 1
-- end --
-- level 1 --
   integer  al              = 5
-- end --
Parsed:
----------------------------------------------------------------------
{
  import "NameOfModule";
  NameOfModule::`ag = (27);
  NameOfModule::`al = (27);
  return NameOfModule::ag;
}
----------------------------------------------------------------------
[Interpreter] tests/module.ycp:25 [YCP] Assignment not possible: variable NameOfModule::al is not declared
[Scanner] tests/module.ycp:38 Syntax error: Only declarations allowed inside a module.
