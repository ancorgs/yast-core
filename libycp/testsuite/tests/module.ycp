{
    module "NameOfModule";

    global integer ag = 1;
    integer al = 2;

    global define NameOfModule() ``{
	y2milestone ("I am a constructor\n");
	al = 5;
    }

    global define af() ``{ return ag+al; }
}

{
  _dump_scope("NameOfModule");

  return NameOfModule::af();
}

{
  import "NameOfModule";

  NameOfModule::ag = 27;
  NameOfModule::`al = 27;

  return NameOfModule::ag;
}

{
   module "A";

   integer a = 1;

   // syntax error, only declarations allowed inside a "module" block
   if (true)
   {
      integer b = 1;
   }

   return a + B::b;
}
