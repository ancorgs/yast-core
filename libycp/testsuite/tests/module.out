(nil)
(6)
(27)
